crates/typst-syntax/src/parser.rs

```diff
@@ -1628,38 +1702,46 @@ impl<'s> Parser<'s> {
         at
     }
 
+    /// Assert that we are at the given SyntaxKind and eat it. This should be used when
+    /// moving between functions that expect to start with a specific token.
     #[track_caller]
     fn assert(&mut self, kind: SyntaxKind) {
-        assert_eq!(self.current, kind);
+        assert_eq!(self.current.kind, kind);
         self.eat();
     }
 
-    fn convert(&mut self, kind: SyntaxKind) {
-        self.current = kind;
+    /// Convert the current token's SyntaxKind and eat it.
+    fn convert_and_eat(&mut self, kind: SyntaxKind) {
+        self.current.kind = kind;
         self.eat();
     }
 
+    /// Whether there was a newline in the previous token.
+    /// Warning: only use this in Markup mode due to `prev_trivia` in Math/Code mode.
+    ///
+    /// TODO idea: maybe emit newline tokens in Code mode based on NewlineMode?
     fn newline(&mut self) -> bool {
         self.lexer.newline()
     }
 
+    /// The column of the current node.
     fn column(&self, at: usize) -> usize {
+        // TODO idea: keep track of current column in the parser to make this faster?
```

Comment:
probably not really in the hot path. but we can experiment with it.

Reply:
I'll leave this for a later date or for someone else if anyone's interested. Should I keep the TODO in the code or remove it?

Reply:
I would prefer to have it removed unless we have some substantial reasoning or testing that this would actually have a noticable performance impact.